staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/configmapleaselock.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourcelock
+
+import (
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// ConfigMapLeaseLock is used for migration between the configmap and the lease
+type ConfigMapLeaseLock struct {
+	ConfigMapLock *ConfigMapLock
+	LeaseLock     *LeaseLock
+}
+
+// Get returns the older election record of the configmap and the lease
+func (cmll *ConfigMapLeaseLock) Get() (*LeaderElectionRecord, error) {
+	configMapLock, err := cmll.ConfigMapLock.Get()
+	if err != nil {
+		return nil, err
+	}
+	leaseLock, err := cmll.LeaseLock.Get()
+	if err != nil {
+		return nil, err
+	}
+	// check if the older resource lock is released
+	if leaseLock.RenewTime.Before(&configMapLock.RenewTime) {
+		return leaseLock, nil
+	}
+	return configMapLock, nil
+}
+
+// Create attempts to create both configmap and lease
+func (cmll *ConfigMapLeaseLock) Create(ler LeaderElectionRecord) error {
+	err := cmll.ConfigMapLock.Create(ler)
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return err
+	}
+	err = cmll.LeaseLock.Create(ler)
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return err
+	}
+	return nil
+}
+
+// Update will update and existing annotation on both two resources.
+func (cmll *ConfigMapLeaseLock) Update(ler LeaderElectionRecord) error {
+	err := cmll.Update(ler)
+	if err != nil {
+		return err
+	}
+	return cmll.LeaseLock.Update(ler)
+}
+
+// RecordEvent in leader election while adding meta-data
+func (cmll *ConfigMapLeaseLock) RecordEvent(s string) {
+	cmll.ConfigMapLock.RecordEvent(s)
```

Comment:
if this is being used to transition between lock impls over multiple releases, would we not record events for both impls here?